torch/package/package_exporter.py

```diff
@@ -468,14 +486,24 @@ def add_dependency(self, module_name: str, dependencies=True):
 
                 # If we are interning this module, we need to retrieve its
                 # dependencies and package those as well.
-                if pattern_info.action == _ModuleProviderAction.INTERN:
+                if action == _ModuleProviderAction.INTERN or action == _ModuleProviderAction.SELECTIVE_INTERN:
                     self._intern_module(module_name, dependencies)
+                if action == _ModuleProviderAction.SELECTIVE_EXTERN:
+                    module_obj = self._import_module(module_name)
+                    is_package = hasattr(module_obj, "__path__")
+                    source = self._get_source_of_module(module_obj)
+                    self.dependency_graph.add_node(
+                    module_name, action=_ModuleProviderAction.SELECTIVE_EXTERN, provided=True, source=source, is_package=is_package)
+                    for interned_module in self._selective_interns[module_name]:
```

Comment:
The flow is ambiguous to me, we are handling selectively interned modules in two places:
1. On L489, where we intern based on matched patterns
2. Here, where we proactively intern everything we find in selective_interns dict entry.

It's not clear to me which modules are supposed to be handled by 1 vs. 2.

Reply:
Ah will add a comment. Basically it's for flows like 
```
_selective_intern(torch, [torch.fb])
save_src_string("import torch.fb; stuff") <= Case 1
save_src_string("import torch; stuff") <= Case 2
```
I feel like you don't want to break those changes, but by converting _ModuleProviderAction.SELECTIVE_INTERN to _ModuleProviderAction.INTERN you solve those cases anyways.